index.php

```diff
@@ -120,29 +121,33 @@ function mysort($a, $b)
                                 <!-- Tree Death Count -->
                                 <td>' . $db[substr($file, 0, -5) . '.sav'][6] . '</td>
 
-                                <!-- Locomotives Owned -->
-                                <td>' . $db[substr($file, 0, -5) . '.sav'][2] . '</td>
-
                                 <!-- Rolling Stock Owned -->
-                                <td>' . $db[substr($file, 0, -5) . '.sav'][3] . '</td>
+                                <td>' . $db[substr($file, 0, -5) . '.sav'][2] . '/' . $db[substr($file, 0, -5) . '.sav'][3] . '</td>
 
                                 <!-- Max Slope -->
                                 <td>' . round($db[substr($file, 0, -5) . '.sav'][4]) . '%</td>
+
                                 <!-- Shared Link -->';
 
-                            // Checks public save folder to see if we can provide a link
-                            $saveCheck = SHELL_ROOT . 'saves/public/' . substr($file, 5, -5) . '.sav';
-                            if (file_exists($saveCheck)) {
-                                $upTime = filemtime($saveCheck);
-                                $timeCheck = time() - $upTime;
-                                if ($timeCheck < 172800) {
-                                    echo '<td><a href="' . WWW_ROOT . 'saves/public/' . substr($file, 5, -5) . '.sav">Link</a></td>';
+                            // Create savefile name from map file
+                            $saveFile = substr($file, 0, -5) .'.sav';
+
+                            // Check to see if savefile exists in public folder and create download link
+                            if (file_exists(SHELL_ROOT.'saves/public/' . $saveFile)) {
+                              $upTime = filemtime(SHELL_ROOT.'saves/public/' . $saveFile);
+                              $timeCheck = time() - $upTime;
+
+                              // Timecheck to remove public link for download
+                                if ($saveFile < $timeCheck) {
+                                    echo '<td><a href="'.WWW_ROOT.'saves/public/'.$saveFile.'">Link</a></td>';
```

Comment:
saveFile is a string, timeCheck is a number.
This comparison makes no sense here